addons/im_livechat/static/src/embed/common/chatbot/chatbot_model.js

```diff
@@ -152,8 +154,19 @@ export class Chatbot extends Record {
             const nextURL = new URL(answer.redirect_link, window.location.href);
             isRedirecting = url.pathname !== nextURL.pathname || url.origin !== nextURL.origin;
         }
+        const redirects = JSON.parse(
+            expirableStorage.getItem("im_livechat.chatbot_redirect") ?? "[]"
+        );
         const targetURL = new URL(answer.redirect_link, window.location.origin);
-        const redirectionAlreadyDone = targetURL.href === location.href;
+        const redirectionAlreadyDone =
+            targetURL.href === location.href || redirects.includes(message.id);
```

Comment:
Note: we should still check if we already are on the target page, otherwise the script will be blocked (the script restarts when the page is reloaded).